setup.py

```diff
@@ -59,19 +59,17 @@
         ],
         "dev": [
             f"apache-airflow=={os.environ.get('AIRFLOW_VERSION', '2.9.1')}",
-            "autoflake==1.7.7",
             "agate==1.7.1",
             "beautifulsoup4",
-            "black==24.4.2",
+            "ruff~=0.4.0",
             "cryptography~=42.0.4",
             "dbt-core",
             "dbt-duckdb>=1.7.1",
             "Faker",
             "google-auth",
             "google-cloud-bigquery",
             "google-cloud-bigquery-storage",
-            "isort==5.10.1",
-            "mypy~=1.8.0",
+            "mypy~=1.10.0",
```

Comment:
Please also note the mypy bump. Let me know if we'd pinned to 1.8.0 for a reason.